tests/authenticator/RESTfulAPI_TokenAuthenticator_Test.php

```diff
@@ -209,4 +214,84 @@ public function testAuthenticate()
       "TokenAuth authentication failure should return a RESTfulAPI_Error"
     );
   }
-}
\ No newline at end of file
+
+  public function testCanLogin(){
+    $auth = $this->getAuthenticator();
+
+    // check login when canLogIn check fails
+    $request = new SS_HTTPRequest(
+      'GET',
+      'api/auth/login',
+      array(
+        'email' => 'balrog@moria.mine',
+        'pwd'   => 'flame.sword'
+      )
+    );
+
+    $result = $auth->login($request);
+
+    $this->assertEquals(
+      $result['code'],
+      RESTfulAPI_TokenAuthenticator::AUTH_CODE_LOGIN_FAIL,
+      "Logging in a member whose `canLogIn` check fails should not be allowed."
+    );
+
+    // check authenticate when canLogIn check fails
+
+    $request = new SS_HTTPRequest(
```

Comment:
Do we really need to do another test here? Since all that changes is the `if` logic in the `canLogInExtesnion`?